plugin/nginx_plus_collectd.py

```diff
@@ -673,24 +697,38 @@ class NginxStatusAgent(object):
     '''
     Helper class for interacting with a single NGINX+ instance.
     '''
-    def __init__(self, status_host=None, status_port=None, username=None, password=None):
+    def __init__(self, status_host=None, status_port=None, username=None, password=None, api_version=None, api_base_path=None):
         self.status_host = status_host or 'localhost'
         self.status_port = status_port or 8080
         self.auth_tuple = (username, password) if username or password else None
+        self.api_version = api_version
+        self.api_base_path = api_base_path
+
+        detected_api_version = self._get_api_version()
```

Comment:
I think you should only detect the version if a version isn't provided via config.

Reply:
In the last pushed commit, we have made the change. Now, we are only detecting the NGINX+ API type (versioned or legacy) if `APIVersion` is not provided by the user.